apicast/src/oauth.lua

```diff
@@ -1,24 +1,34 @@
-local get_token = require 'get_token'
-local callback = require 'authorized_callback'
-local authorize = require 'authorize'
-
+local env = require 'resty.env'
 local router = require 'router'
+local oauth
 
 local _M = {
-  version = '0.0.1'
+  _VERSION = '0.0.2'
 }
 
+function _M.new()
+  local keycloak = env.get('RHSSO_ENDPOINT')
+  if keycloak then
+    oauth = require 'oauth.keycloak'
```

Comment:
not sure if this is the correct way to do it, but I wanted to only require the correct oauth module, i.e require keycloak if we're using keycloak and apicast_oauth if we're using the built in functionality